src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -128,21 +131,19 @@ public WorkflowJob(ItemGroup parent, String name) {
 
     @Override public void onLoad(ItemGroup<? extends Item> parent, String name) throws IOException {
         super.onLoad(parent, name);
+
         if (buildMixIn == null) {
             buildMixIn = createBuildMixIn();
         }
         buildMixIn.onLoad(parent, name);
-        if (triggers == null) {
-            triggers = new DescribableList<Trigger<?>,TriggerDescriptor>(this);
-        } else {
-            triggers.setOwner(this);
-        }
-        for (Trigger t : triggers) {
-            t.start(this, Items.currentlyUpdatingByXml());
+        if (triggers != null && !triggers.isEmpty()) {
+            setTriggers(triggers.toList());
```

Comment:
Only affects the one-time load after upgrading this plugin, but still—for some plugins the value of `newInstance` may be important.